texts/0006-tauri-releases-schedule.md

```diff
@@ -0,0 +1,53 @@
+- Feature Name: tauri_release_schedule
+- Start Date: 2024-6-6
+- RFC PR: [tauri-apps/rfcs#14](https://github.com/tauri-apps/rfcs/pull/14)
+- Tracking Issue:
+
+# Summary
+
+Define a release schedule for tauri main repo, including crates, NPM packages and CLIs. 
+
+# Motivation
+
+As it stands now, there is no release schedule and releases are made either whenever one of the working group feels like it (mainly Lucas and myself (Amr)),
+or when there is a need for an urgent fix.
+
+As a tauri user or contributor, there is no indication when a release might happen and if will it include a certain feature
+one might be waiting for which leads me to either try to figure how to use unreleased features directly from source or wait indefinitely.   
+
+A release schedule should make the releasing process more predictable for maintainers and users and prevents the tauri project from ever appearing stagnant.
+
+# Explanation
+
+We follow semver versioning for all tauri crates, NPM packages and CLIs and so we have 6 type of releases we are ever gonna make for stable releases,
+here is what each release type cycle would look like:
+
+- `patch`: weekly
+  - `patch` versions can be released when needed, in the case of an urgent bug fix or security vulnerability. 
+  - PRs with `minor` bumps shall be marked as `Ready for next minor` and will not be merged until the next `minor` release cycle.
```

Comment:
Won't this increase the risk of conflicts for a 2 months period? If the number of branches is not a problem, a new branch could be created after checking out the latest minor release tag, then pull request for patches could target this new branch instead?

I don't know how much the risk exists on Tauri and I still have limited experience, but I know it happened to me quite a few times in a similar configuration and it always was some time lost and stability risk increase to handle all the conflicts at once. It just requires a strong enough branch policy to always know how to handle each situation and avoid bottlenecks, and to always be able to target the root of latest major and minor releases.

Reply:
The conflict is indeed a risk here and usually once a PR is approved for next minor, it will be the responsibility of the maintainers to fix the conflict and merge these PRs before release.

Reply:
It seems like the issue wouldn't be just be conflicting with patches that are merged in the intermediate period, but rather with other 'Ready for next minor' PRs.

I'd recommend a read through of '[branching for release](https://trunkbaseddevelopment.com/branch-for-release/)' from Trunk Based Development. While it may not be an overall approach that Tauri wishes to follow it will potentially highlight some of the issues that are often encountered which may help feed into designing the release process here.

Reply:
For what it's worth, I agree with @MGough that a trunk style release would be more beneficial for this kind of consistent release schedule.

In practical terms, this would just reverse the process, instead of having all minor changes marked for "ready for minor", after a minor a release, the codebase would be branched into a "patch" branch, and then minor PRs are merged as into main, and then anything people want to be included in a patch, would be tagged with "ready for patch", and then cherrypicked into the "patch" branch, and then when the next minor release happens, main is moved over to patch again, and cycle starts over.

Reply:
I don't like this either to be honest, I would probably prefer creating a new branch for every minor release and having everything merged to main branch and cherrypick them to minor release tracking branch for patch releases

Reply:
I also know some users like to directly use new features from GitHub directly which usually helps catch bugs faster before they are released so I guess I am in favor of releasing patches through cherry picking into another branch.

Reply:
Changed this in [`b3af691` (#14)](https://github.com/tauri-apps/rfcs/pull/14/commits/b3af691dbbfb4402d7ae445f5562efe231436906), let me know if this sounds alright with yall.